pkg/controller/services.go

```diff
@@ -1455,58 +1460,125 @@ func (cont *AciController) infraRtAttEntPDeleted(dn string) {
 	}
 }
 
+func (cont *AciController) vpcIfDeleted(dn string) {
+	var deleted bool
+	cont.indexMutex.Lock()
+	for tDn, hostInfo := range cont.hostFabricPathDnMap {
+		if _, present := hostInfo.vpcIfDn[dn]; present {
+			cont.log.Info("Deleting vpcIf, dn :", dn)
+			delete(hostInfo.vpcIfDn, dn)
+			if len(hostInfo.vpcIfDn) == 0 {
+				cont.log.Infof("Removing fabricPathDn(%s) of ipg : %s ", hostInfo.fabricPathDn, hostInfo.host)
+				hostInfo.fabricPathDn = ""
+				deleted = true
+			}
+			cont.hostFabricPathDnMap[tDn] = hostInfo
```

Comment:
Nit: Shouldn't you delete this mapping itself at this point. As long as its handled in updateDeviceCluster, its ok.